mqtt/mqtt-bridge/tests/integration.rs

```diff
@@ -479,3 +479,77 @@ async fn bridge_forwards_messages_after_restart() {
     upstream_client.shutdown().await;
     local_client.shutdown().await;
 }
+
+#[tokio::test]
+async fn recreate_upstream_bridge_when_fails() {
```

Comment:
It would help if we can write up the scenario as a comment for the tests, similar like we do in `mqtt-broker/tests/compliance.rs`. Even if we can get the idea from the name, with comments it is easier to understand how test is working, what clients are being used, what ops performed and what is expected.